### DISCUSSION TITLE
While using Quote(") as delimeter throwing error

### DISCUSSION THREAD

**Discussion by snspvd** [JoshClose/CsvHelper#1695] [Q&A]
While working with **quote(")** as delimeter throwing error (It Is not supported).but it working fine for ",","|"
```cs
public DataTable ParseCsvFile(StreamReader csvData, string delimeter = SamplicityConstants.Etl.Delimiter)
        {

            if (csvData.Peek() != -1)
            {
                using (var csv = new CsvReader(csvData))
                {
                    csv.Configuration.Delimiter = delimeter;
                    using (var dr = new CsvDataReader(csv))
                    {
                        var dt = new DataTable();
                        dt.Columns.Add(new DataColumn()
                        {
                            ColumnName = "RowIndex",
                            AutoIncrement = true,
                            AutoIncrementStep = 1,
                            AutoIncrementSeed = 1
                        });
                        dt.Load(dr);
                        return dt;
                    }
                }
            }
            return null;
        }
```

**Comment by pramod-md** [comment]
https://stackoverflow.com/questions/66061980/should-quote-should-not-qote-headers-should-remain-as-is

**Comment by JoshClose** [comment]
You can do it if the delimiter is different than the escape and quote.

**Comment by JoshClose** [comment]
Example:
```cs
void Main()
{
	var s = new StringBuilder();
	s.Append("Id\"Name\r\n");
	s.Append("1\"one\r\n");
	var config = new CsvConfiguration(CultureInfo.InvariantCulture)
	{
		Delimiter = "\"",
		Quote = '\'',
		Escape = '\''
	};
	using (var reader = new StringReader(s.ToString()))
	using (var csv = new CsvReader(reader, config))
	{
		csv.GetRecords<Foo>().ToList().Dump();
	}
}

private class Foo
{
	public int Id { get; set; }
	public string Name { get; set; }
}
```

**Reply by pramod-md** [reply]
This is what my program is the data right now is static, it will be dynamic with 100 records and need 
1. Deliminiter "|"        ==> Everywhere on headers and also the row data
2. Should Quote        ==> Only on row data

Below works for me, but only the header is wrong because should quotes are added there which I don't want in headers.
Rest all is good
Output - 

"Id"|"Name"
"1"|"one"
"2"|"two"

The output I need is -

 Id | Name
"1"|"one"
"2"|"two"          


         var records = new List<Foo>
            {
                new Foo { Id = 1, Name = "one" },
                new Foo { Id = 2, Name = "two" },
            };
           
            var config = new CsvConfiguration(CultureInfo.InvariantCulture)
            {
                Delimiter = "|",
                ShouldQuote = new ShouldQuote((field, fieldType, context) => true)
            };

            var stream = new MemoryStream();
            using (var writeFile = new StreamWriter(stream, leaveOpen: true))
            {
                
                var csv = new CsvWriter(writeFile, config);
                await csv.WriteRecordsAsync(records);
            }

            stream.Position = 0; //reset stream
            return File(stream, "application/octet-stream", "Reports.csv");

**Reply by pramod-md** [reply]
+ CsvHelper Version - 22.1.2

**Reply by JoshClose** [reply]
Change it to only quote when not on the header row then.

```cs
ShouldQuote = (field, type, context) => context.Row > 1
```

**Reply by pramod-md** [reply]
The above worked for me thanks, 

    is there a way to achieve this
    
    “The”||”Brown”|”Fox”|”Jumped”|”Over”|”The”||”Dog”
     
    instead of this
    
    “The”|""|”Brown”|”Fox”|”Jumped”|”Over”|”The”|""|”Dog” 

 i.e. if a value is null, do not include the double quotes?
